dynamic_rest/filters.py

```diff
@@ -517,7 +518,15 @@ def _filter_queryset(
         )
 
         if query:
-            queryset = queryset.filter(query)
+            # Convert internal django ValidationError to APIException-based one
+            # in order to resolve validation errors from 500 status code to
+            # 400.
+            try:
+                queryset = queryset.filter(query)
+            except InternalValidationError as e:
+                raise ValidationError(
+                    dict(e) if hasattr(self, 'error_dict') else list(e)
```

Comment:
Right; but here it says `hasattr(self, 'error_dict')`, where `self` is the `DynamicFilterBackend` object.